src/OrchardCore/OrchardCore.Modules/ModularTenantContainerMiddleware.cs

```diff
@@ -36,8 +37,14 @@ public class ModularTenantContainerMiddleware
 
             var shellSettings = _runningShellTable.Match(httpContext);
 
+            if (shellSettings?.State == TenantState.Initializing)
```

Comment:
Why not move this block under `if (shellSettings != null)`

503 is a nice addition instead of 404 (which I assume is happening without this change). Shouldn't we also add a retry-after header then? For instance with a 1 second delay?